lib/main/atom/views/semanticView.tsx

```diff
@@ -0,0 +1,424 @@
+import atomConfig = require("../atomConfig");
+import atomUtils = require("../utils");
+import * as sp from "atom-space-pen-views";
+import * as view from "./view";
+import {clientResolver} from "../../atomts"
+import * as dom from "jsx-render-dom"
+import {TypescriptBuffer} from '../../typescriptBuffer';
+
+/**
+ * Eventually
+ */
+namespace rts {
+  /** 0 based length */
+  export function indent(indent: number) {
+    return Array(indent + 1).join().split('').map(i => "\u00a0\u00a0\u00a0\u00a0");
+  }
+}
+
+//TODO extract?
+interface Position {
+  line: number;
+  offset: number;
+}
+
+interface SemanticTreeNode {
+  text: string;
+  kind: string;
+  kindModifiers: string;
+  spans: Array<{start: Position, end: Position}>;
+  childItems?: Array<SemanticTreeNode>;
+}
+
+export interface TypescriptFileBuffer {
+  buffer: TypescriptBuffer;
+  isOpen: boolean;
+}
+
+class SemanticViewRenderer {
+
+  editor: AtomCore.IEditor;
+  navTree: SemanticTreeNode | null;
+  getTypescriptBuffer: (filePath: string) => Promise<TypescriptFileBuffer>;
+  root: HTMLElement;
+  selectedNode: HTMLElement | null;
+
+  constructor(options: {getTypescriptBuffer: (filePath: string) => Promise<TypescriptFileBuffer>}) {
+    this.getTypescriptBuffer = options.getTypescriptBuffer;
+    this.navTree = null;
+    this.selectedNode = null;
+  }
+
+  setEditor(editor: AtomCore.IEditor) {
+    this.editor = editor;
+  }
+
+  setNavTree(navTree: SemanticTreeNode | null) {
+    this.navTree = navTree;
+    this._render();
+  }
+
+  forceUpdate() {
+    //FIXME should this be done differently?
+    this._render();
+  }
+
+  loadNavTree(buffer: TypescriptFileBuffer, filePath?: string) {
+    filePath = filePath ? filePath : this.editor.getPath();
+    buffer.buffer.getClient(filePath).then(client => client.executeNavTree({file: filePath as string}).then(navtreeResult => {
```

Comment:
Like I mentioned elsewhere, you just need to get the client here which is something that `ClientResolver` can do for you. Basically you have to replace `getTypescriptBuffer` with an instance of `ClientResolver` provided by `atomts.ts`.